owncloudData/src/test/java/com/owncloud/android/data/appRegistry/datasources/implementation/OCRemoteAppRegistryDataSourceTest.kt

```diff
@@ -0,0 +1,166 @@
+import com.owncloud.android.domain.appregistry.model.AppRegistry
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.resources.appregistry.responses.AppRegistryResponse
+import com.owncloud.android.lib.resources.appregistry.services.OCAppRegistryService
+import com.owncloud.android.testutil.OC_ACCOUNT_NAME
+import com.owncloud.android.testutil.OC_FILE
+import com.owncloud.android.utils.createRemoteOperationResultMock
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class OCRemoteAppRegistryDataSourceTest {
+
+    private lateinit var ocRemoteAppRegistryDataSource: OCRemoteAppRegistryDataSource
+    private val clientManager: ClientManager = mockk(relaxed = true)
+    private val ocAppRegistryService: OCAppRegistryService = mockk()
+    private val appResgitryResponse = AppRegistryResponse(value = mockk(relaxed = true))
+    private val openWebEndpoint = "https://example.com"
+
+    private val appName = "TestApp"
+
+    @Before
+    fun init() {
+        every { clientManager.getAppRegistryService(any()) } returns ocAppRegistryService
+
+        ocRemoteAppRegistryDataSource = OCRemoteAppRegistryDataSource(clientManager)
+    }
+
+    @Test
+    fun `getAppRegistryForAccount returns the appRegistry object`() {
+        val getAppRegistryResult: RemoteOperationResult<AppRegistryResponse> =
+            createRemoteOperationResultMock(data = appResgitryResponse, isSuccess = true)
+
+        val appResgitryMock = AppRegistry(accountName = OC_ACCOUNT_NAME, mimetypes = emptyList())
+
+        every { ocAppRegistryService.getAppRegistry() } returns getAppRegistryResult
+
+        val appRegistry = ocRemoteAppRegistryDataSource.getAppRegistryForAccount(OC_ACCOUNT_NAME)
+
+        assertEquals(appResgitryMock, appRegistry)
+
+        verify(exactly = 1) { ocAppRegistryService.getAppRegistry() }
+    }
+
+    @Test(expected = Exception::class)
+    fun `getAppRegistryForAccount returns an Exception when the operation is not successful`() {
+        val getAppRegistryResult: RemoteOperationResult<AppRegistryResponse> =
+            createRemoteOperationResultMock(data = appResgitryResponse, isSuccess = false)
+
+        every { ocAppRegistryService.getAppRegistry() } returns getAppRegistryResult
+
+        ocRemoteAppRegistryDataSource.getAppRegistryForAccount(OC_ACCOUNT_NAME)
+
+    }
+
+    @Test(expected = Exception::class)
+    fun `getAppRegistryForAccount returns an Exception when getAppRegistry() has an error controlled by an Exception`() {
+        every { ocAppRegistryService.getAppRegistry() } throws Exception()
+
+        ocRemoteAppRegistryDataSource.getAppRegistryForAccount(OC_ACCOUNT_NAME)
+    }
+
+    @Test
+    fun `getUrlToOpenInWeb returns an url to open in website`() {
+        val expectedUrl = "https://example.com/file123/TestApp"
```

Comment:
There is no constant defined to use here? If there isn't I will create e new one.

Reply:
I didn't find any.